MTV-1433: Creating source and destination providers, separated by provider (#567)
kubev2v/forklift-documentation: documentation/modules/creating-migration-plan-2-8.adoc

+
The *Create migration plan* wizard opens to the *Select source provider* interface.
. Select the source provider of the VMs you want to migrate.
+
The *Select virtual machines* interface opens.
. Select the VMs you want to migrate and click *Next*.
endif::[]

ifdef::vms[]
= Creating a migration plan starting with specific VMs

You can create a migration plan based on specific VMs, starting on the *Providers for virtualization* page. Note the specific options for migrations from VMware or {rhv-short} providers.

.Procedure

. In the {ocp} web console, click *Providers for virtualization*.
. In the row of the appropriate source provider, click *VMs*.
+
The *Virtual Machines* tab opens.
. Select the VMs you want to migrate and click *Create migration plan*.
endif::[]
+
The *Create migration plan* pane opens. It displays the source provider's name and suggestions for a target provider and namespace, a network map, and a storage map.
. Enter the *Plan name*.
. Make any needed changes to the editable items.
. Click *Add mapping* to edit a suggested network mapping or a storage mapping, or to add one or more additional mappings.
. Click *Create migration plan*.
+

> {project-short} validates the migration plan and the *Plan details* page opens, indicating whether the plan is ready for use or contains an error. The details of the plan are listed, and you can edit the items you filled in on the previous page. If you make any changes, {project-short} validates the plan again.

Reviewer: Please mention the "Settings" with the cold/warm migration.
Or is there a different place for it?

Author: @mnecas Good question. Investigating this.

. VMware source providers only (All optional):

* *Preserving static IPs of VMs*: By default, virtual network interface controllers (vNICs) change during the migration process.  As a result, vNICs that are configured with a static IP linked to the interface name in the guest VM lose their IP. To avoid this, click the Edit icon next to *Preserve static IPs* and toggle the *Whether to preserve the static IPs* switch in the window that opens. Then click *Save*.
+
{project-short} then issues a warning message about any VMs for which vNIC properties are missing. To retrieve any missing vNIC properties, run those VMs in vSphere in order for the vNIC properties to be reported to {project-short}.

> * *Entering a list of decryption passphrases for disks encrypted using Linux Unified Key Setup (LUKS)*: To enter a list of decryption passphrases for LUKS-encrypted devices, in the *Settings* section, click the Edit icon next to *Disk decryption passphrases*, enter the passphrases, and then click *Save*. You do not need to enter the passphrases in a specific order - for each LUKS-encrypted device, {project-short} tries each passphrase until one unlocks the device.

Reviewer 2: ```suggestion
* *Entering a list of decryption passphrases for disks encrypted using Linux Unified Key Setup (LUKS)*: To enter a list of decryption passphrases for LUKS-encrypted devices, in the *Settings* section, click the *Edit* icon next to *Disk decryption passphrases*, enter the passphrases, and then click *Save*. You do not need to enter the passphrases in a specific order.  For each LUKS-encrypted device, {project-short} tries each passphrase until a passphrase unlocks the device.
```

Reviewer 3: You should implement Shruti's suggestion. Using `-` is not good

* *Specifying a root device*: Applies to multi-boot VM migrations only. By default, {project-short} uses the first bootable device detected as the root device.
+

> To specify a different root device, in the *Settings* section, click the Edit icon next to *Root device* and choose a device from the list of commonly-used options, or enter a device in the text box.

Reviewer 2: ```suggestion
To specify a different root device, in the *Settings* section, click the *Edit* icon next to *Root device* and choose a device from the list of commonly-used options, or enter a device in the text box.
```

Author: Good catch. (Change made in https://github.com/kubev2v/forklift-documentation/pull/585).

> To specify a different root device, in the *Settings* section, click the Edit icon next to *Root device* and choose a device from the list of commonly-used options, or enter a device in the text box.

Reviewer 3: ```suggestion
To specify a different root device, in the *Settings* section, click the Edit icon next to *Root device* and choose a device from the list of commonly used options, or enter a device in the text box.
```
Using a hyphen is acceptable. However, it is not necessary in a compound modifier in which the first word is an adverb that ends in `ly`.

+
{project-short} uses the following format for disk location: `/dev/sd<disk_identifier><disk_partition>`. For example, if the second disk is the root device and the operating system is on the disk's second partition, the format would be: `/dev/sdb2`. After you enter the boot device, click *Save*.
+

> If the conversion fails because the boot device provided is incorrect, it is possible to get the correct information by looking at the conversion pod logs.

Reviewer 2: ```suggestion
If the conversion fails because the boot device provided is incorrect, it is possible to get the correct information by checking the conversion pod logs.
```

Author: @RichardHoch change in PR about plans.

Author: Good catch. (Change made in https://github.com/kubev2v/forklift-documentation/pull/585).

. {rhv-short} source providers only (Optional):

* *Preserving the CPU model of VMs that are migrated from {rhv-short}*: Generally, the CPU model (type) for {rhv-short} VMs is set at the cluster level, but it can be set at the VM level, which is called a custom CPU model.

> By default, {project-short} sets the CPU model on the destination cluster as follows: {project-short} preserves custom CPU settings for VMs that have them, but, for VMs without custom CPU settings, {project-short} does not set the CPU model. Instead, the CPU model is later set by {virt}.

Reviewer 2: ```suggestion
By default, {project-short} sets the CPU model on the destination cluster as follows: 
* {project-short} preserves custom CPU settings for VMs that have them.
* For VMs without custom CPU settings, {project-short} does not set the CPU model. Instead, the CPU model is later set by {virt}.
```

Author: @RichardHoch Fix in PR about plans.

Author: @shdeshpa07 Much better. (Change made in https://github.com/kubev2v/forklift-documentation/pull/585).

> By default, {project-short} sets the CPU model on the destination cluster as follows: {project-short} preserves custom CPU settings for VMs that have them, but, for VMs without custom CPU settings, {project-short} does not set the CPU model. Instead, the CPU model is later set by {virt}.

Reviewer 3: ```suggestion
By default, {project-short} sets the CPU model on the destination cluster as follows: {project-short} preserves custom CPU settings for VMs that have them, but for VMs without custom CPU settings, {project-short} does not set the CPU model. Instead, the CPU model is later set by {virt}.
```

+

> To preserve the cluster-level CPU model of your {rhv-short} VMs, in the *Settings* section, click the Edit icon next to *Preserve CPU model*. Toggle the *Whether to preserve the CPU model* switch, and then click *Save*.

Reviewer 2: ```suggestion
To preserve the cluster-level CPU model of your {rhv-short} VMs, in the *Settings* section, click the *Edit* icon next to *Preserve CPU model*. Toggle the *Whether to preserve the CPU model* switch, and then click *Save*.
```

Author: @shdeshpa07  Good catch. (Change made in https://github.com/kubev2v/forklift-documentation/pull/585).
